community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/PlannerCache.scala

```diff
@@ -43,34 +43,34 @@ class PlannerFactory(graph: GraphDatabaseQueryService, kernelAPI: KernelAPI, ker
 
 
   def create(spec: PlannerSpec_v2_3) =  spec.planner match {
-    case CypherPlanner.rule => CompatibilityFor2_3Rule(graph, as2_3(config), Clock.SYSTEM_CLOCK, kernelMonitors, kernelAPI)
-    case _ => CompatibilityFor2_3Cost(graph, as2_3(config),
-                                      Clock.SYSTEM_CLOCK, kernelMonitors, kernelAPI, log, spec.planner, spec.runtime)
+    case CypherPlanner.rule =>
+      v2_3.RuleCompatibility(graph, as2_3(config), Clock.SYSTEM_CLOCK, kernelMonitors, kernelAPI)
+    case _ =>
+      v2_3.CostCompatibility(graph, as2_3(config), Clock.SYSTEM_CLOCK, kernelMonitors, kernelAPI, log, spec.planner, spec.runtime)
   }
 
   def create(spec: PlannerSpec_v3_1) = spec.planner match {
-    case CypherPlanner.rule => CompatibilityFor3_1Rule(graph, as3_1(config), CypherCompiler.CLOCK, kernelMonitors, kernelAPI)
-    case _ => CompatibilityFor3_1Cost(graph, as3_1(config),
-                                      CypherCompiler.CLOCK, kernelMonitors, kernelAPI, log, spec.planner, spec.runtime,
-                                      spec.updateStrategy)
+    case CypherPlanner.rule =>
+      v3_1.RuleCompatibility(graph, as3_1(config), CypherCompiler.CLOCK, kernelMonitors, kernelAPI)
+    case _ =>
+      v3_1.CostCompatibility(graph, as3_1(config), CypherCompiler.CLOCK, kernelMonitors, kernelAPI, log, spec.planner, spec.runtime, spec.updateStrategy)
   }
 
   def create(spec: PlannerSpec_v3_2) = spec.planner match {
-    case CypherPlanner.rule => CompatibilityFor3_2Rule(graph, config, CypherCompiler.CLOCK, kernelMonitors, kernelAPI)
-    case _ => CompatibilityFor3_2Cost(graph, config, CypherCompiler.CLOCK, kernelMonitors, kernelAPI, log, spec.planner,
-      spec.runtime, spec.updateStrategy)
+    case CypherPlanner.rule =>
+      v3_2.RuleCompatibility(graph, config, CypherCompiler.CLOCK, kernelMonitors, kernelAPI)
+    case _ =>
+      v3_2.CostCompatibility(graph, config, CypherCompiler.CLOCK, kernelMonitors, kernelAPI, log, spec.planner, spec.runtime, spec.updateStrategy)
   }
-
 }
 
 class PlannerCache(factory: PlannerFactory)  {
-  private val cache_v2_3 = new mutable.HashMap[PlannerSpec_v2_3, CompatibilityFor2_3]
-  private val cache_v3_1 = new mutable.HashMap[PlannerSpec_v3_1, CompatibilityFor3_1]
-  private val cache_v3_2 = new mutable.HashMap[PlannerSpec_v3_2, CompatibilityFor3_2]
+  private val cache_v2_3 = new mutable.HashMap[PlannerSpec_v2_3, v2_3.Compatibility]
+  private val cache_v3_1 = new mutable.HashMap[PlannerSpec_v3_1, v3_1.Compatibility]
+  private val cache_v3_2 = new mutable.HashMap[PlannerSpec_v3_2, v3_2.Compatibility]
```

Comment:
By just looking at the code - these are very small maps that are not emptied. I think it makes sense to move from our LRU caches to Caffeine, but this is not the same thing.